src/transformers/image_transforms.py

```diff
@@ -707,3 +708,96 @@ def convert_to_rgb(image: ImageInput) -> ImageInput:
 
     image = image.convert("RGB")
     return image
+
+
+def bilinear_interpolation(image: np.ndarray, y: float, x: float):
+    # docstyle-ignore
+    """
+    A bilinear interpolation of the estimated values of the `image` at non integer indexes `y` and `x`.
+    
+    
+    Original Image at                    Original Image at
+      x_1, y_1                             x_1, y_2 
+        +---+                               +---+       
+        | +-|-------------------------------|-+ |       
+        +---+                               +---+       
+            |                                   |         
+            |             Pixel at (x,y) where  |         
+            |             x and y non integers  |         
+            |                     +---+         |         
+            |                     |   |         |         
+            |                     +---+         |         
+        +---+                               +---+       
+        | +-|-------------------------------|-+ |       
+        +---+                               +---+       
+                                                        
+    Original Image at                    Original Image at
+      x_1, y_2                             x_2, y_2 
+    
+    The estimated value of the pixel is computed using the following equation :
+    
+    $$ \text{Image}_{x,y} = \frac{1}{(x_1 - x_2)(y_2-y_1)} \begin{bmatrix} x_2 - x & x - x_1\end{bmatrix}
+    \begin{bmatrix} \text{Image}_{x_1,y_1} & \text{Image}_{x_2,y_1}\\ \text{Image}_{x_1,y_2} & \text{Image}_{x_2,y_2}\\
+    \end{bmatrix} \begin{bmatrix} y_2 - y \\ y-y_2\end{bmatrix} $$
+    
+    For more details about bilinear interplation, see [on the wikipedia
+    page](https://en.wikipedia.org/wiki/Bilinear_interpolation)
+    """
+    height = image.shape[0]
+    width = image.shape[1]
+
+    x1 = max(min(math.floor(x), width - 1), 0)
+    y1 = max(min(math.floor(y), height - 1), 0)
+    x2 = max(min(math.ceil(x), width - 1), 0)
+    y2 = max(min(math.ceil(y), height - 1), 0)
+
+    a = image[y1, x1]
+    b = image[y2, x1]
+    c = image[y1, x2]
+    d = image[y2, x2]
+
+    dx = x - x1
+    dy = y - y1
+
+    new_pixel = a * (1 - dx) * (1 - dy)
+    new_pixel += b * dy * (1 - dx)
+    new_pixel += c * dx * (1 - dy)
+    new_pixel += d * dx * dy
+    return new_pixel
+
+
+def np_bilinear_resize(image: np.ndarray, new_height: int, new_width: int):
+    """
+    Taken from `[here](https://stackoverflow.com/questions/70024313/resize-using-bilinear-interpolation-in-python)`
+    this is the equivalent of the `torchvision.transforms.Resize(size=[chunk_frames, self.feature_size])`. This
+    function is not optimal in terms of performances, but has the same results as `torchvision` counterpart when called
```

Comment:
Lol, not optimal? You don't say! I think it takes less time to run through the actual model than do a resize this way.